extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/management/ManagementInterfaceBuildTimeConfig.java

```diff
@@ -0,0 +1,68 @@
+package io.quarkus.vertx.http.runtime.management;
+
+import java.util.OptionalInt;
+
+import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.annotations.ConfigPhase;
+import io.quarkus.runtime.annotations.ConfigRoot;
+import io.quarkus.vertx.http.runtime.AuthConfig;
+import io.vertx.core.http.ClientAuth;
+
+/**
+ * Management interface configuration.
+ */
+@ConfigRoot(name = "management", phase = ConfigPhase.BUILD_AND_RUN_TIME_FIXED)
+public class ManagementInterfaceBuildTimeConfig {
+
+    public AuthConfig auth;
```

Comment:
The problem with referencing `AuthConfig` is that it includes not only `Basic` but also `FormAuth` - if you'd like it supported with the management console then it is a cross-port interaction between the main and management endpoints which will most likely require CORS enablement - which is not a problem but it will need to tested to confirm it works.

The other things that `AuthConfig` allows is to configure HTTP policies (authentication, roles, etc) which if supported have to be tested. `proactiveAuthentication` is also there - not sure if configuring it is needed for the management interface case. 

IMHO having `ManagementAuthConfig` will be better, starting with the basic auth enablement only for a start, as it will allow us evolve it independently from the main `AuthConfig` setup, since the management and application endpoints security requirements will most likely be different. There is also a pending task to try and make `AuthConfig` runtime configurable and having Management depending on it can make it more difficult, though it is of lesser concern.

Reply:
If a users wants to use for ex `Form Auth` to have a secured access to both main and management endpoints then the current main `AuthConfig` should be enough I guess. It will need to be tested but it won't be a problem of this PR.

Reply:
That's a good point, I should remove the AuthConfig for now. 

MAnagement interface will use a different authentication support (in a future PR, as it's rarely used).

Reply:
Removed. Auth will be handled in another PR and will use vert.x auth directly.